lib/policy/policyValidator.js

```diff
@@ -24,33 +26,39 @@ const errDict = {
 };
 
 // parse ajv errors and return early with the first relevant error
-function _parseErrors(ajvErrors) {
-    // deep copy is needed as we have to assign custom error description
-    const parsedErr = Object.assign({}, errors.MalformedPolicyDocument);
-    parsedErr.description = 'Syntax errors in policy.';
+function _parseErrors(ajvErrors, policyType) {
+    let parsedErr;
+    if (policyType === 'user') {
+        // deep copy is needed as we have to assign custom error description
+        parsedErr = Object.assign({}, errors.MalformedPolicyDocument);
+        parsedErr.description = 'Syntax errors in policy.';
+    }
+    if (policyType === 'resource') {
+        parsedErr = Object.assign({}, errors.MalformedPolicy);
+    }
     ajvErrors.some(err => {
         const resource = err.dataPath;
         const field = err.params ? err.params.missingProperty : undefined;
         const errType = err.keyword;
         if (errType === 'type' && (resource === '.Statement' ||
-            resource === '.Statement.Resource' ||
-            resource === '.Statement.NotResource')) {
+            resource.includes('.Resource') ||
```

Comment:
I changed these to use `.includes` so that policies that are arrays of multiple statements would still be caught. In that case the `resource` would look like `.Statement[0].Resource`, for example